packages/astro/src/transitions/router.ts

```diff
@@ -462,7 +462,25 @@ async function transition(
 		const init: RequestInit = {};
 		if (preparationEvent.formData) {
 			init.method = 'POST';
-			init.body = preparationEvent.formData;
+			// Form elements without enctype explicitly set default to application/x-www-form-urlencoded.
+			// In order to maintain compatibility with Astro 4.x, we need to check the value of enctype
+			// on the attributes property rather than accessing .enctype directly. Astro 5.x may
+			// introduce defaulting to application/x-www-form-urlencoded as a breaking change, and then
+			// we can access .enctype directly.
+			//
+			// Note: getNamedItem can return null in real life, even if TypeScript doesn't think so, hence
+			// the ?.
+			const form =
+				preparationEvent.sourceElement instanceof HTMLFormElement
+					? preparationEvent.sourceElement
+					: preparationEvent.sourceElement instanceof HTMLElement &&
+					    'form' in preparationEvent.sourceElement
+					  ? (preparationEvent.sourceElement.form as HTMLFormElement)
+					  : preparationEvent.sourceElement?.closest('form');
```

Comment:
I'm open to better ideas here than `'form' in element`. I wanted to be able to use `.form` and not rely on `closest('form')` since people can set the `form` attribute. This ternary falls back to `closest('form')` though I can't think of a scenario off the top of my head where that would happen.

Reply:
I think that is really good!